[Typography foundations] Add `README` for new Text component (#6715)
Shopify/polaris: polaris-react/src/components/Text/README.md

---
name: Text
category: Titles and text
keywords:
  - titles
  - text
  - typographic
  - spacing
  - display
  - heading
  - body
  - success
  - critical
  - warning
  - subdued
  - regular
  - medium
  - semibold
  - bold
  - list
status:
  value: Alpha
  message: This component is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution.
---

# Text

> A component for displaying Text

Reviewer: Just a thought, maybe we should align this to wording in the alpha release guide Figma.

```suggestion
Typography helps establish hierarchy and communicate important content by creating clear visual patterns.
```

Author: I was looking for that!

Reviewer: got it from [here](https://www.figma.com/file/lywHPxKoSsqzhQyasj6VON/Typography-foundations?node-id=1663%3A61376)

---

## Examples

> ### Display Small

Reviewer: nit:

```suggestion
### Display small
```

Use to create a small display text.

```jsx
<Text variant="displaySm">Good evening, Dominic.</Text>
```

> ### Display Medium

Reviewer: nit:

```suggestion
### Display medium
```

Use to create a medium display text.

```jsx
<Text variant="displayMd">Good evening, Dominic.</Text>
```

> ### Display Large

Reviewer: nit:

```suggestion
### Display large
```

Use to create a large display text.

```jsx
<Text variant="displayLg">Good evening, Dominic.</Text>
```

> ### Heading Small

Reviewer: nit:

```suggestion
### heading small
```

Use to create a small heading text.

```jsx
<Text variant="headingSm" as="h6">
  Online store dashboard
</Text>
```

> ### Heading Medium

Reviewer: nit:

```suggestion
### Heading medium
```

Use to create a medium heading text.

```jsx
<Text variant="headingMd" as="h5">
  Online store dashboard
</Text>
```

> ### Heading Large

Reviewer: nit:

```suggestion
### Heading large
```

Use to create a large heading text.

```jsx
<Text variant="headingLg" as="h4">
  Online store dashboard
</Text>
```

> ### Heading Extra Large

Reviewer: nit:

```suggestion
### Heading extra large
```

Use to create an extra large heading text.

```jsx
<Text variant="headingXl" as="h3">
  Online store dashboard
</Text>
```

> ### Body Small

Reviewer: nit:

```suggestion
### Body small
```

Use to create a small body text.

```jsx
<Text variant="bodySm" as="p">
  Shopify POS is the easiest way to sell your products in person. Available for
  iPad, iPhone, and Android.
</Text>
```

> ### Body Medium

Reviewer: nit:

```suggestion
### Body medium
```

Use to create a medium body text.

```jsx
<Text variant="bodyMd" as="p">
  Shopify POS is the easiest way to sell your products in person. Available for
  iPad, iPhone, and Android.
</Text>
```

> ### Body Large

Reviewer: nit:

```suggestion
### Body large
```

Use to create a large body text.

```jsx
<Text variant="bodyLg" as="p">
  Shopify POS is the easiest way to sell your products in person. Available for
  iPad, iPhone, and Android.
</Text>
```

> ### Align Inherit

Reviewer: ```suggestion
### With align inherit
```

Use to inherit parent alignment.

```jsx
<Text variant="bodyLg" as="p" alignment="inherit">
  Manage your Shopify store on-the-go with real-time notifications, access to
  your dashboard, and order management, all from your smartphone.
</Text>
```

> ### Align Start

Reviewer: ```suggestion
### With align start
```

Use to align text at start of horizontal line.

```jsx
<Text variant="bodyLg" as="p" alignment="start">
  Manage your Shopify store on-the-go with real-time notifications, access to
  your dashboard, and order management, all from your smartphone.
</Text>
```

> ### Align Center

Reviewer: ```suggestion
### With align center
```

Use to align text at center of horizontal line.

```jsx
<Text variant="bodyLg" as="p" alignment="center">
  Manage your Shopify store on-the-go with real-time notifications, access to
  your dashboard, and order management, all from your smartphone.
</Text>
```

> ### Align End

Reviewer: ```suggestion
### With align end
```

Use to align text at end of horizontal line.

```jsx
<Text variant="bodyLg" as="p" alignment="end">
  Manage your Shopify store on-the-go with real-time notifications, access to
  your dashboard, and order management, all from your smartphone.
</Text>
```

> ### Align Justify

Reviewer: ```suggestion
### With align justify
```

Use to align text on a vertical line.

```jsx
<Text variant="bodyLg" as="p" alignment="justify">
  Manage your Shopify store on-the-go with real-time notifications, access to
  your dashboard, and order management, all from your smartphone.
</Text>
```

> ### Font Weight Regular

Reviewer: ```suggestion
### With font weight regular
```

Use to give text a regular font weight.

```jsx
<Text variant="bodyMd" as="p" fontWeight="regular">
  Sales this year
</Text>
```

> ### Font Weight Medium

Reviewer: ```suggestion
### With font weight medium
```

Use to give text a medium font weight.

```jsx
<Text variant="bodyMd" as="p" fontWeight="medium">
  Sales this year
</Text>
```

> ### Font Weight Semi-bold
> 
> Use to give text a semi-bold font weight.

Reviewer: ```suggestion
### With font weight semibold

Use to give text a semibold font weight.
```

```jsx
<Text variant="bodyMd" as="p" fontWeight="semibold">
  Sales this year
</Text>
```

> ### Font Weight Bold

Reviewer: ```suggestion
### With font weight bold
```

Use to give text a bold font weight.

```jsx
<Text variant="bodyMd" as="p" fontWeight="bold">
  Sales this year
</Text>
```

> ### Success text color

Reviewer: ```suggestion
### With color success
```

Use in combination with a symbol showing an increasing value to indicate an upward trend.

```jsx
<Text variant="bodyMd" as="p" color="success">
  Orders increased
</Text>
```

> ### Critical text color

Reviewer: ```suggestion
### With color critical
```

Use in combination with a symbol showing a decreasing value to indicate a downward trend.

```jsx
<Text variant="bodyMd" as="p" color="critical">
  Orders decreased
</Text>
```

> ### Warning text color

Reviewer: ```suggestion
### With color warning
```

Use to denote something that needs attention, or that merchants need to take action on.

```jsx
<Text variant="bodyMd" as="p" color="warning">
  Scheduled maintenance
</Text>
```

> ### Subdued text color

Reviewer: ```suggestion
### With color subdued
```
